src/main/java/com/commercetools/sync/commons/utils/CommonTypeUpdateActionUtils.java

```diff
@@ -0,0 +1,30 @@
+package com.commercetools.sync.commons.utils;
+
+
+import io.sphere.sdk.commands.UpdateAction;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+import java.util.Optional;
+
+public class CommonTypeUpdateActionUtils {
+
+    /**
+     * Compares two {@link Object} and returns a supplied {@link UpdateAction} as a result in an
+     * {@link Optional}. If no update action is needed, for example in case where both the {@link Object}
+     * have the same values, an empty {@link Optional} is returned.
+     *
+     * @param oldObject    the object which should be updated.
+     * @param newObject    the object with the new information.
+     * @param updateAction the update action to return in the optional.
+     * @return A filled optional with the update action or an empty optional if the object values are identical.
+     */
+    @Nonnull
+    public static <T> Optional<UpdateAction<T>> buildUpdateAction(@Nullable final Object oldObject,
+                                                                  @Nullable final Object newObject,
+                                                                  @Nullable final UpdateAction<T> updateAction) {
```

Comment:
this is one of the main benefits of functional programming - lazy initialization.